pkg/clusteragent/admission/mutate/autoinstrumentation/auto_instrumentation_test.go

```diff
@@ -37,6 +37,169 @@ import (
 
 const commonRegistry = "gcr.io/datadoghq"
 
+func TestInjectAutoInstruConfigV2(t *testing.T) {
+	tests := []struct {
+		name                  string
+		pod                   *corev1.Pod
+		expectedInjectorImage string
+		libsToInject          []libInfo
+		wantErr               bool
+	}{
+		{
+			name: "no libs, no injection",
+			pod:  common.FakePod("java-pod"),
+		},
+		{
+			name:                  "nominal case: java",
+			pod:                   common.FakePod("java-pod"),
+			expectedInjectorImage: commonRegistry + "/apm-injector:v27",
+			libsToInject: []libInfo{
+				java.libInfo("", "gcr.io/datadoghq/dd-lib-java-init:v1"),
+			},
+		},
+		{
+			name:                  "nominal case: java & python",
+			pod:                   common.FakePod("java-pod"),
+			expectedInjectorImage: commonRegistry + "/apm-injector:v27",
+			libsToInject: []libInfo{
+				java.libInfo("", "gcr.io/datadoghq/dd-lib-java-init:v1"),
+				python.libInfo("", "gcr.io/datadoghq/dd-lib-python-init:v1"),
+			},
+		},
+		{
+			name: "java + injector-tag-override",
+			pod: common.FakePodSpec{
+				Annotations: map[string]string{
+					"admission.datadoghq.com/apm-injector.version": "v0",
+				},
+			}.Create(),
+			expectedInjectorImage: commonRegistry + "/apm-injector:v0",
+			libsToInject: []libInfo{
+				java.libInfo("", "gcr.io/datadoghq/dd-lib-java-init:v1"),
+			},
+		},
+		{
+			name: "java + injector-image-override",
+			pod: common.FakePodSpec{
+				Annotations: map[string]string{
+					"admission.datadoghq.com/apm-injector.custom-image": "docker.io/library/apm-inject-package:v27",
+				},
+			}.Create(),
+			expectedInjectorImage: "docker.io/library/apm-inject-package:v27",
+			libsToInject: []libInfo{
+				java.libInfo("", "gcr.io/datadoghq/dd-lib-java-init:v1"),
+			},
+		},
+	}
+
+	wmeta := fxutil.Test[workloadmeta.Component](
```

Comment:
Should workloadmeta be initialized before each test? Otherwise, its state might carry over between tests, which could be confusing even if it isn't causing any tests to fail at the moment.